src/openrct2/peep/Guest.cpp

```diff
@@ -1585,30 +1587,34 @@ bool Guest::DecideAndBuyItem(Ride* ride, int32_t shopItem, money32 price)
         if (itemValue < price)
         {
             itemValue -= price;
-            if (shopItem == SHOP_ITEM_UMBRELLA)
+            if (shopItem == SHOP_ITEM_UMBRELLA && climate_is_raining())
             {
-                if (climate_is_raining())
-                    goto loc_69B221;
+                //goto loc_69B221; now using below variable to skip all the code
+                //until the place where loc_69B221 was;
+                isRainingAndUmbrella = true;
             }
 
-            itemValue = -itemValue;
-            if (Happiness >= 128)
-                itemValue /= 2;
+            if (isRainingAndUmbrella == false)
```

Comment:
It's a bit weird to compare a `bool` to a `bool` literal and it's not how we do it in the code base. Just do:
`!isRainingAndUmbrella`

Reply:
Yes, that makes sense!